src/modules/dashboard/components/EditDomainDialog/EditDomainDialogForm.tsx

```diff
@@ -0,0 +1,217 @@
+import PermissionRequiredInfo from '~core/PermissionRequiredInfo';
+
+import { Colony, useLoggedInUser } from '~data/index';
+import { useTransformer } from '~utils/hooks';
+
+import { getAllUserRoles } from '../../../transformers';
+import { canArchitect } from '../../../users/checks';
+
+import { FormValues } from './EditDomainDialog';
+import styles from './EditDomainDialogForm.css';
+
+const MSG = defineMessages({
+  titleEdit: {
+    id: 'dashboard.EditDomainDialog.EditDomainDialogForm.titleEdit',
+    defaultMessage: 'Edit team details',
+  },
+  name: {
+    id: 'dashboard.EditDomainDialog.EditDomainDialogForm.name',
+    defaultMessage: 'Team name',
+  },
+  domain: {
+    id: 'dashboard.EditDomainDialog.EditDomainDialogForm.domain',
+    defaultMessage: 'Select domain',
+  },
+  purpose: {
+    id: 'dashboard.EditDomainDialog.EditDomainDialogForm.name',
+    defaultMessage: 'What is the purpose of this team?',
+  },
+  annotation: {
+    id: 'dashboard.EditDomainDialog.EditDomainDialogForm.annotation',
+    defaultMessage: 'Explain why you’re editing this team',
+  },
+  noPermission: {
+    id:
+      // eslint-disable-next-line max-len
+      'dashboard.EditDomainDialog.EditDomainDialogForm.noPermission',
+    defaultMessage:
+      // eslint-disable-next-line max-len
+      'You need the {roleRequired} permission in {domain} to take this action.',
+  },
+});
+
+interface Props {
+  back?: () => void;
+  colony: Colony;
+  isSubmitting;
+  isValid;
+}
+
+const EditDomainDialogForm = ({
+  back,
+  colony,
+  colony: { domains },
+  handleSubmit,
+  isSubmitting,
+  isValid,
+  setFieldValue,
+  values: { domainId },
+}: Props & FormikProps<FormValues>) => {
+  const [domainColor, setDomainColor] = useState(Color.LightPink);
+
+  const { walletAddress, username, ethereal } = useLoggedInUser();
+  const allUserRoles = useTransformer(getAllUserRoles, [colony, walletAddress]);
+
+  const hasRegisteredProfile = !!username && !ethereal;
+  const canEditDomain = hasRegisteredProfile && canArchitect(allUserRoles);
+
+  const domainOptions = useMemo(
+    () =>
+      sortBy(
+        domains.map(({ name, ethDomainId }) => ({
+          value: ethDomainId.toString(),
+          label: name,
+        })),
+        ['value'],
+      ),
+
+    [domains],
+  );
+
+  const handleDomainChange = (selectedDomainId) => {
+    const selectedDomain = domains.find(
+      (domain) => domain.ethDomainId.toString() === selectedDomainId,
+    );
+    if (selectedDomain) {
+      setFieldValue(
+        'domainId',
+        selectedDomain.ethDomainId.toString() || undefined,
+      );
+      setFieldValue('domainColor', selectedDomain.color || undefined);
+      setFieldValue('domainName', selectedDomain.name || undefined);
+      setFieldValue('domainPurpose', selectedDomain.description || undefined);
+      setDomainColor(selectedDomain.color);
+    }
+  };
+  useEffect(() => {
+    if (domainId) {
+      handleDomainChange(domainId);
+    }
+  }, []);
```

Comment:
This has missing dependencies: `domainId` and `handleDomainChange`

Reply:
Nope, i want fire this only once, on first render

Reply:
When you pass empty array as second parameter of useEffect it will behave like componentDidMount.

Reply:
I know that, I just don't understand why do you only need this to run at first render.

But it really doesn't matter this is a minor nit-pick, also I trust that you did the right thing :)